src/drivers/emulated/serial.c

```diff
@@ -0,0 +1,74 @@
+#include "serial.h"
+#include <stdint.h>
+
+// Function to read a byte from the specified port
+static inline uint8_t inb(uint16_t port) {
```

Comment:
inb and outb are part of the x86 specific io device, it has to be under the arch/x86 folder